agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -286,22 +287,22 @@ public void start() {
         try {
             _connection.start();
         } catch (final NioConnectionException e) {
-            logger.warn("NIO Connection Exception  " + e);
+            logger.warn("NIO Connection Exception {}", e);
```

Comment:
a `warn()` with an exception and then an `info()` with a more generic message. I think we can improve more here.
i.e:
- warn with `e.getLocalizedMessage()` and
- remove the `info()` which is of lesser level anyway.